tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -1543,5 +1543,121 @@ int main(int argc, char **argv)
+
+        /* Set a resumption psk */
+        DEFER_CLEANUP(struct s2n_stuffer psk_identity = { 0 }, s2n_stuffer_free);
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&psk_identity, 0));
+        EXPECT_OK(s2n_setup_encrypted_ticket(client_conn, &psk_identity));
+        struct s2n_offered_psk_list identity_list = { .conn = client_conn };
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&identity_list.wire_data, 0));
+        EXPECT_OK(s2n_write_test_identity(&identity_list.wire_data, &psk_identity.blob));
+        EXPECT_OK(s2n_select_resumption_psk(server_conn, &identity_list));
+
+        /* Add arbitrary amount to ticket age so a nonzero value implies it's been processed */
+        client_conn->psk_params.chosen_psk->ticket_age_add = 10;
+
+        /* ClientHello 1 */
+        EXPECT_SUCCESS(s2n_client_hello_send(client_conn));
+
+        /* Read the obfuscated ticket age from ClientHello 1 */
+        EXPECT_SUCCESS(s2n_stuffer_skip_read(&client_conn->handshake.io, 342));
+        uint32_t obfuscated_ticket_age_1 = 0;
+        EXPECT_SUCCESS(s2n_stuffer_read_uint32(&client_conn->handshake.io, &obfuscated_ticket_age_1));
+
+        /* Read the binder from ClientHello 1 */
+        EXPECT_SUCCESS(s2n_stuffer_skip_read(&client_conn->handshake.io, 3));
+        uint8_t binder_1[32] = { 0 };
+        EXPECT_SUCCESS(s2n_stuffer_read_bytes(&client_conn->handshake.io, binder_1, 32));
+        EXPECT_SUCCESS(s2n_stuffer_reread(&client_conn->handshake.io));
+
+        /* Ensure that the ticket age and binder was updated after ClientHello 1 */
+        EXPECT_TRUE(obfuscated_ticket_age_1 != 0);
+        uint8_t zero_array[32] = { 0 };
+        EXPECT_FALSE(s2n_constant_time_equals(binder_1, zero_array, 32));
+
+        EXPECT_SUCCESS(s2n_stuffer_copy(&client_conn->handshake.io, &server_conn->handshake.io,
+                                        s2n_stuffer_data_available(&client_conn->handshake.io)));
+
+        /* Server receives ClientHello 1 */
+        EXPECT_SUCCESS(s2n_client_hello_recv(server_conn));
+        EXPECT_SUCCESS(s2n_set_connection_hello_retry_flags(server_conn));
+
+        /* Server sends HelloRetryRequest */
+        EXPECT_SUCCESS(s2n_server_hello_retry_send(server_conn));
+
+        EXPECT_SUCCESS(s2n_stuffer_wipe(&client_conn->handshake.io));
+        EXPECT_SUCCESS(s2n_stuffer_copy(&server_conn->handshake.io, &client_conn->handshake.io,
+                                        s2n_stuffer_data_available(&server_conn->handshake.io)));
+        client_conn->handshake.message_number = 1; /* hello retry */
+
+        /* Client receives HelloRetryRequest */
+        EXPECT_SUCCESS(s2n_server_hello_recv(client_conn));
+
+        EXPECT_SUCCESS(s2n_stuffer_rewrite(&client_conn->handshake.io));
+
+        /* Add another arbitrary amount to ticket age so a change in the value implies it's been processed */
+        client_conn->psk_params.chosen_psk->ticket_age_add = 20;
```

Comment:
Without this, the generated obfuscated ticket age is 0, so it can't be determined if this was actually updated in the second client hello. I think this is because the ticket age is set to the current time, and the same current time is retrieved when calculating the ticket age. Setting `ticket_age_add` adds the value to the ticket age, which forces the resulting ticket age to be something other than 0, indicating that a new ticket age has been calculated and written.

Reply:
Right okay. I changed this to set the ticket issue time instead, since a change in the ticket issue time should definitely change the resulting obfuscated ticket age. Let me know if this makes more sense: 9eae76fd3fc8fb4126ef6f6fc154ca1617ce156f